qcodes/instrument/base_representation.py

```diff
@@ -0,0 +1,104 @@
+""" Representation methods for instruments """
+import logging
+import uuid
+from collections import namedtuple
+from functools import partial
+from html import escape
+from typing import Any, Dict, Optional
+
+from xarray.core.formatting_html import _icon, _mapping_section, _obj_repr
```

Comment:
@peendebak these seem to be private functions from deep inside xarray package, we can't have this as a dependency in qcodes because it's not guaranteed that these will not change (or that xarray will notify the world about changing those), because they are private (the ``_`` prefix in the name).  Hence this must change. I suggest that there functions get re-implemented right here in qcodes (the direct copy-pasting is not an option because xarray and qcodes licenses are different and it may be more trouble to deal with that, than just reimplement those functions), would you be so kind to do that? after that the PR can be merged (perhaps on time for the mid-May release!)